### ISSUE TITLE
Investigate whether GH Actions can be used for build verification and automatic releases

### ISSUE THREAD

**Issue by emmaffiondavies** [andidavies92/pokemon-rby-dx#5]
## Overview

If possible, we'd like to use GH Actions to perform build validation (possibly even run some kind of automated tests in future too) and automatically publish releases when the version number of the game is incremented. Conduct a practical investigation to determine whether Actions could be used for these purposes. Only do rapid prototyping though, as the actual implementation (if possible) will be done in #2.

## Purpose

Having automated build verification would help keep the `main` branch clean and automatic releases would just be a modern nice-to-have as a process that then wouldn't need to be done by hand each time.

## Tasks

* [x] Try to create a proof-of-concept build pipeline in Actions.
* [x] If successful, try to publish the output from the build stage as built artefacts.
* [x] If successful, try to package and publish these artefacts as a test release.
* [x] Write-up findings and clean-up any temporary generated stuff.

## Resources

* https://www.github.com/features/actions
* https://docs.github.com/en/free-pro-team@latest/actions

## Future

* [#2 &ndash; Initial repository-tidying refactors](https://www.github.com/AmateurPanda92/pokemon-rby-dx/issues/2)

**Comment by emmaffiondavies** [comment]
This all seems pretty trivial. The quick 'n' dirty CI workflow I put together (starting from a template) for build validation is below, and the [Create a Release](https://www.github.com/marketplace/actions/create-a-release), [Upload a Build Artifact](https://www.github.com/marketplace/actions/upload-a-build-artifact) and [Upload a Release Asset](https://www.github.com/marketplace/actions/upload-a-release-asset) first-party actions in the Actions Marketplace should cover the rest (plus the usual logic for pre-verifying pull requests, only publishing on new Git tag, etc.):

```yaml
# This is a basic workflow to help you get started with Actions

name: CI

# Controls when the action will run. 
on:
  # Triggers the workflow on push or pull request events but only for the main branch
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]

  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

# A workflow run is made up of one or more jobs that can run sequentially or in parallel
jobs:
  # This workflow contains a single job called "build"
  build:
    # The type of runner that the job will run on
    runs-on: ubuntu-latest

    # Steps represent a sequence of tasks that will be executed as part of the job
    steps:
      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
      - uses: actions/checkout@v2

      # Runs a single command using the runners shell
      - name: Run a one-line scripts
        run: echo Hello, world!

      # Runs a set of commands using the runners shell
      - name: Run a multi-line script
        run: |
          sudo apt-get install make git gcc pkg-config flex bison libpng-dev
          git clone -b v0.4.0 --depth=1 https://github.com/rednex/rgbds
          sudo make -C rgbds install
          make
          ls
```
